projects/i18n/switch/language-switcher.ts

```diff
@@ -0,0 +1,53 @@
+import {Inject, Injectable, Optional} from '@angular/core';
+import {TuiLanguage, TuiLanguageName} from '@taiga-ui/i18n/interfaces';
+import {TUI_DEFAULT_LANGUAGE} from '@taiga-ui/i18n/tools';
+import {BehaviorSubject, Observable, of} from 'rxjs';
+
+import {TUI_LANGUAGE_CHUNK_LOADER} from './language-chunk-loader-token';
+import {tuiAsyncLoadLanguage} from './load-lang';
+import {
+    TUI_LANGUAGE_DEFAULT_STORAGE,
+    TUI_LANGUAGE_STORAGE_KEY,
+    TuiLangChunkLoader,
+} from './options';
+
+// @dynamic
+@Injectable({providedIn: `root`})
+export class TuiLanguageSwitcher extends BehaviorSubject<Observable<TuiLanguage>> {
+    constructor(
+        @Inject(TUI_DEFAULT_LANGUAGE)
+        private readonly fallback: TuiLanguage,
+        @Inject(TUI_LANGUAGE_STORAGE_KEY)
+        private readonly key: string,
+        @Inject(TUI_LANGUAGE_DEFAULT_STORAGE)
+        private readonly storage: Storage,
+        @Optional()
+        @Inject(TUI_LANGUAGE_CHUNK_LOADER)
+        private readonly loader: TuiLangChunkLoader | null,
+    ) {
+        super(tuiAsyncLoadLanguage({key, storage, fallback, loader}));
+    }
+
+    get currentLanguage(): TuiLanguageName {
+        return (this.storage.getItem(this.key) as TuiLanguageName) ?? this.fallback.name;
```

Comment:
Empty string doesn't make sense as `currentLanguage` so let's change `??` to `||` for simplicity and smaller transpiled bundle.

Reply:
fixed